software/apps/romi_control/main.c

```diff
@@ -211,109 +336,12 @@ int main(void) {
-            lsm9ds1_start_gyro_integration();
-            state = TURNING;
-        } else {
-          state = WAITING;
-          kobukiDriveDirect(0, 0);
-        }
-        break;
-      }
-      case TURNING: {
-        float current_angle = lsm9ds1_read_gyro_integration().z_axis;
-        float diff = waypoint[1] - current_angle;
-        if (is_button_pressed(&sensors)) {
-          lsm9ds1_stop_gyro_integration();
-          state = OFF;
-        } else if (fabs(diff) < angle_threshold) {
-            lsm9ds1_stop_gyro_integration();
-            encoder_prev_left = sensors.leftWheelEncoder;
-            encoder_prev_right = sensors.rightWheelEncoder;
-            state = DRIVING;
-        }
-        else {
-          state = TURNING;
-          int8_t sign = (2 * (diff > 0)) - 1;
-          int16_t speed = sign * fmax(.8 * fabs(diff), 50);
-          kobukiDriveDirect(-speed, speed);
-        }
-        break;
-      }
-      case DRIVING: {
-        float diff_left = waypoint[0] - total_distance_left;
-        float diff_right = waypoint[0] - total_distance_right;
-        float wheel_diff = total_distance_left - total_distance_right;
-        if (is_button_pressed(&sensors)) {
-            total_distance_left = 0;
-            total_distance_right = 0;
-            //total_distance = 0;
-          state = OFF;
-        } else if ((fabs(diff_left) < distance_threshold) && (fabs(diff_right) < distance_threshold)) {
-            total_distance_left = 0;
-            total_distance_right = 0;
-            acknowledged = 0;
-            simple_ble_notify_char(&ack_char);
-            //total_distance = 0;
-            state = WAITING;
-        } else {
-          state = DRIVING;
-          encoder_cur_left = sensors.leftWheelEncoder;
-          encoder_cur_right = sensors.rightWheelEncoder;
-          total_distance_left += measure_distance(encoder_cur_left, encoder_prev_left);
-          total_distance_right += measure_distance(encoder_cur_right, encoder_prev_right);
-          //total_distance = (.5 * total_distance_left) + (.5 * total_distance_right);
-          encoder_prev_left = encoder_cur_left;
-          encoder_prev_right = encoder_cur_right;
-          int8_t sign_left = (2 * (diff_left > 0)) - 1;
-          //Wheel diff might not work if backwards :(
-          int gain = 200;
-          int16_t speed_left =  sign_left * fmax(gain * fabs(diff_left), 80) - 250 * wheel_diff;
-          //int16_t speed_left =  sign_left * fmax(-220 * wheel_diff, 40);
-          int8_t sign_right = (2 * (diff_right > 0)) - 1;
-          int16_t speed_right = sign_right * fmax(gain * fabs(diff_right), 80) + 250 * wheel_diff;
-          //speed_left = 100;
-          //speed_right = 100;
-          kobukiDriveDirect(speed_left, speed_right);
-          //kobukiDriveDirect(speed_left , sign_right * 80);
-        }
-        break;
-      }
-    }
+  system_state_t system_state = init_state();
+  while(true) {
+    inputs_t inputs = get_inputs();
```

Comment:
I remember there being a sleep call for 1 ms in the labs to prevent some sort of error (may only be for printing to debug, not 100% sure). might want to keep that

Reply:
Added, thanks